components/media-upload/index.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * WordPress dependencies
+ */
+import withFilters from '../higher-order/with-filters';
+
+/**
+ * This is a placeholder for the media upload component necessary to make it possible to provide
+ * an integration with the core blocks that handle media files. By default it renders nothing but
+ * it provides a way to have it overridden with the `blocks.MediaUpload` filter.
+ *
+ * @return {WPElement} Media upload element.
+ */
+const MediaUpload = () => null;
+
+export default withFilters( 'components.MediaUpload' )( MediaUpload );
```

Comment:
I'm not sure we want to move components relying on filters to the `components` folder. I feel like the components folder is for generic React components while adding filters adds an unwanted dependency `the wp hooks`.

I guess it's probably not the first component doing this, but I wanted to share this concern of mine.

Reply:
@aduth @mtias @mcsf, what is your take on this? I'm fine with leaving it where it was if you disagree with the move. All the docs I added are still valuable :)

Reply:
In general, I see this is a block tool, so placing it in the blocks folder makes sense. Not sure how that will evolve, but I expect most block API UI items to be there.

Reply:
Thanks for leaving your feedback 👍